### DISCUSSION TITLE
Monorepo ideas and first feedback

### DISCUSSION THREAD

**Discussion by matzeeable** [envkey/envkey#62] [Ideas and feedback]
Hey!

I have now started to migrate our  `.env` to Envkey within a larger monorepos. During the migration I have already noticed something and would like to leave feedback about this. But first of all, I would like to say thank you for this brilliant tool, it is super structured, and it just works!

### Avoid Duplication

We work with a kind of microservices in our monorepo and each backend runs within its own backend (Docker). These microservices are also partly connected and communicate with each other. Currently, some keys (e.g. REST API Secrets) are duplicated via blocks and assigned to the respective backends as an app:

![image](https://user-images.githubusercontent.com/1008534/236137597-dd3ecead-57dd-4039-86c1-16c97a1516b7.png)

Of course, we could also create a lot of blocks and assign them to the respective apps. In the end, however, I think we would have ~5 blocks per app that contain only one variable (we also need to ensure variable isolation and Apps should only get variables which they really consume). 

#### Solution 1: Granularity share variables to apps

Currently, blocks can be assigned to specific apps. This makes sense, for example, when external services are involved. However, if we want to share an internal secret within backends, as already described above, this is only possible via a block, which in turn is possible via a block with just one variable due to the fact that we would like to consume variables isolated. A feature to share only one variable with an app would be great.

#### Solution 2: Mark duplicate values

As soon as a value is duplicated within all apps and blocks, it would make sense to mark them accordingly in the UI and, if necessary, also show where they were duplicated.

#### Solution 3: Global search

Since we have duplicated Secrets in the current state, it would make sense to offer a global search with which one can also search for values. As soon as we rotate a secret (later possibly also programmatically), we do not have to click through all apps and blocks and adjust the values individually.

#### Drag & drop variables

If we restructure our environment variables in the future, it would take a lot of time to move variables. Maybe a feature to move variables by drag & drop would make sense.

#### Block folders

As you can see in the screenshot above, we currently prefix blocks with "Apps:", "Server:", "Service:", "S3:", and so on. Folders for blocks would provide an overview here.

#### Disable account recovery on new device as basic user

In this way, we want to prevent our employees from accessing our secrets via a private PC (not a company PC).

**Comment by danenania** [comment]
Thanks so much for the thoughtful feedback @matzeeable!

**Reply by matzeeable** [reply]
Added another feedback "Disable account recovery on new device as basic user".
